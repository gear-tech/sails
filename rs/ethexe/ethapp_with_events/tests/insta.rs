#[test]
fn ethapp_with_events_generate_idl() {
    let mut idl = Vec::new();
    sails_rs::generate_idl::<ethapp_with_events::MyProgram>(&mut idl).unwrap();
    let idl = String::from_utf8(idl).unwrap();
    insta::assert_snapshot!(idl);
}
