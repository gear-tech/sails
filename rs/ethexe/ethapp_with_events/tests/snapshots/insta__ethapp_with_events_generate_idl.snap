---
source: ethapp_with_events/tests/insta.rs
expression: idl
---
constructor {
  Create : ();
};

service Svc1 {
  DoThis : (p1: u32, p2: str) -> u32;
  query This : (p1: bool) -> bool;

  events {
    DoThisEvent: struct {
      /// Some u32 value
      /// #[indexed]
      p1: u32,
      p2: str,
    };
    ThisEvent;
  }
};

service Svc2 {
  DoThis : (p1: u32, p2: str) -> u32;
};
